vertical-pod-autoscaler/pkg/admission-controller/Makefile

```diff
@@ -1,33 +1,59 @@
 all: build
 
-TAG?=v0.0.1
-REGISTRY?=k8s.gcr.io
+TAG?=dev
+REGISTRY?=staging-k8s.gcr.io
 FLAGS=
 ENVVAR=
 GOOS?=linux
+COMPONENT=admission-controller
+FULL_COMPONENT=vpa-${COMPONENT}
 
 deps:
 	go get github.com/tools/godep
 
 build: clean deps
 	$(ENVVAR) GOOS=$(GOOS) godep go build ./...
-	$(ENVVAR) GOOS=$(GOOS) godep go build -o admission-controller
+	$(ENVVAR) GOOS=$(GOOS) godep go build -o ${COMPONENT}
+
+build-binary: clean deps
+	$(ENVVAR) GOOS=$(GOOS) godep go build -o ${COMPONENT}
 
 test-unit: clean deps build
 	$(ENVVAR) godep go test --test.short -race ./... $(FLAGS)
 
-docker:
+docker-build:
+ifndef REGISTRY
+	ERR = $(error REGISTRY is undefined)
+	$(ERR)
+endif
+ifndef TAG
+	ERR = $(error TAG is undefined)
+	$(ERR)
+endif
+	docker build --pull -t ${REGISTRY}/${FULL_COMPONENT}:${TAG} .
+
+docker-push:
 ifndef REGISTRY
 	ERR = $(error REGISTRY is undefined)
 	$(ERR)
 endif
-	docker build --pull -t ${REGISTRY}/vpa-admission-controller:${TAG} .
-	docker push ${REGISTRY}/vpa-admission-controller:${TAG}
+ifndef TAG
+	ERR = $(error TAG is undefined)
+	$(ERR)
+endif
+	docker push ${REGISTRY}/${FULL_COMPONENT}:${TAG}
+
+docker-builder:
+	docker build -t vpa-autoscaling-builder ../../builder
+
+build-in-docker: clean docker-builder
+	docker run -v `pwd`/../..:/gopath/src/k8s.io/autoscaler/vertical-pod-autoscaler vpa-autoscaling-builder:latest bash -c 'cd /gopath/src/k8s.io/autoscaler/vertical-pod-autoscaler && make build-binary -C pkg/admission-controller'
```

Comment:
Is it intentional to not parametrize `admission-controller` here?

Reply:
Looking at the rest of the Makefiles, I guess my question is more if there's any difference between them except this? And why parametrize them using COMPONENT if they're forked anyway?

Reply:
No specific reason I guess I can inline everywhere if that's more readable

Reply:
Ack, as you prefer